dolphin/api/v1/storages.py

```diff
@@ -27,7 +51,53 @@ def show(self, req, id):
         return dict(name="Storage 2")
 
     def create(self, req, body):
-        return dict(name="Storage 3")
+        """
+        This function for registering the new storage device
+        :param req:
+        :param body: "It contains the all input parameters"
+        :return:
+        """
+        # Check if body is valid
+        if not self.is_valid_body(body, 'storages'):
+            msg = ("Storage entity not found in request body")
+            raise exc.HTTPUnprocessableEntity(explanation=msg)
+
+        storage = body['storages']
+
```

Comment:
from your code here, the api json body should be like this:
```
{
"storages": {
}
}
```

Reply:
yes, right. this may be useful when we give multiple inputs in same json file. though openstack also uses same.